ts/src/cryptocom.ts

```diff
@@ -1609,6 +1611,47 @@ export default class cryptocom extends Exchange {
         return this.extend (request, params);
     }
 
+    /**
+     * @method
+     * @name cryptocom#editOrder
+     * @description edit a trade order
+     * @see https://exchange-docs.crypto.com/exchange/v1/rest-ws/index.html#private-amend-order
+     * @param {string} id order id
+     * @param {string} symbol unified market symbol of the order to edit
+     * @param {string} [type] not used by cryptocom editOrder
+     * @param {string} [side] not used by cryptocom editOrder
+     * @param {float} amount (mandatory) how much of the currency you want to trade in units of the base currency
+     * @param {float} price (mandatory) the price for the order, in units of the quote currency, ignored in market orders
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @param {string} [params.clientOrderId] the original client order id of the order to edit, required if id is not provided
+     * @returns {object} an [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+     */
+    async editOrder (id: string, symbol: string, type: OrderType, side: OrderSide, amount: Num = undefined, price: Num = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = this.editOrderRequest (id, symbol, amount, price, params);
+        const response = await this.v1PrivatePostPrivateAmendOrder (request);
+        const result = this.safeDict (response, 'result', {});
+        return this.parseOrder (result);
+    }
+
+    editOrderRequest (id: string, symbol: string, amount: number, price: Num = undefined, params = {}) {
+        const request: Dict = {};
+        if (id !== undefined) {
+            request['order_id'] = id;
+        } else {
+            const originalClientOrderId = this.safeStringN (params, [ 'orig_client_oid', 'client_oid', 'origClientOid', 'clientOrderId', 'client_order_id' ]);
```

Comment:
@yzh-pelle why are we reading so many keys? we should read only the exchange-specific and the ccxt unified